MtfhReportingDataListener.Tests/E2ETests/Steps/TenureCreatedUseCaseSteps.cs

```diff
@@ -0,0 +1,100 @@
+using Amazon.Glue;
+using Amazon.Lambda.SQSEvents;
+using Avro.Generic;
+using Confluent.Kafka;
+using Confluent.Kafka.SyncOverAsync;
+using Confluent.SchemaRegistry.Serdes;
+using FluentAssertions;
+using Hackney.Shared.Tenure.Boundary.Response;
+using MtfhReportingDataListener.Gateway;
+using MtfhReportingDataListener.Infrastructure.Exceptions;
+using MtfhReportingDataListener.Factories;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+using System.Linq;
+
+namespace MtfhReportingDataListener.Tests.E2ETests.Steps
+{
+    public class TenureCreatedUseCaseSteps : BaseSteps
+    {
+        public SQSEvent.SQSMessage TheMessage { get; private set; }
+
+
+        public TenureCreatedUseCaseSteps()
+        {
+            _eventType = EventTypes.TenureCreatedEvent;
+        }
+
+        public async Task WhenTheFunctionIsTriggered(Guid id, IGlueFactory glue)
+        {
+            TheMessage = await TriggerFunction(id, glue).ConfigureAwait(false);
+        }
+
+        public void ThenEntityNotFoundExceptionIsThrown(Guid id)
+        {
+            _lastException.Should().NotBeNull();
+            _lastException.Should().BeOfType(typeof(EntityNotFoundException<TenureResponseObject>));
+            (_lastException as EntityNotFoundException<TenureResponseObject>).Id.Should().Be(id);
+        }
+        public void ThenTheCreatedDataIsSavedToKafka(string schemaDefinition, TenureResponseObject tenure)
+        {
+            var consumerconfig = new ConsumerConfig
+            {
+                BootstrapServers = Environment.GetEnvironmentVariable("DATAPLATFORM_KAFKA_HOSTNAME"),
+                GroupId = "4c659d6b-4739-4579-9698-a27d1aaa397d",
```

Comment:
Does that need to be hardcoded or can we use a random Guid value here?

Reply:
We cannot create a random Guid as the Group ID needs to specify which consumer group the consumer is a member of (hence in this case the value should be the same in all three tests so I can retrieve the correct message)
More info can be found here: https://docs.confluent.io/clients-confluent-kafka-dotnet/current/overview.html#initialization